src/node_api.cc

```diff
@@ -67,13 +67,16 @@ void node_napi_env__::EnqueueFinalizer(v8impl::RefTracker* finalizer) {
 }
 
 void node_napi_env__::DrainFinalizerQueue() {
-  // As userland code can delete additional references in one finalizer,
-  // the list of pending finalizers may be mutated as we execute them, so
-  // we keep iterating it until it is empty.
+  // Process all pending finalizers. If new finalizers are added during
+  // Finalize() calls,
+  // they are handled in subsequent recursive calls.
```

Comment:
nit: `in subsequent recursive calls.` - please update the comment. The latest code does not use the recursion anymore. This is great BTW! :)

Reply:
I'm so happy to hear this. Thanks so much for the suggestion